design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,357 @@
+  Sites. Some fields and edges are valid for Mutable or Immutable OSes.
+  - `install_packages`: list of packages to be installed during the update. 
+    This field is used only for Mutable OSes.
+  - `update_sources`: 'DEB822 Source Format' for Debian style OSs, contains 
+    the sources of the packages to be installed during the update, if new 
+    sources needs to be added.
+  - `update_policy`: enum field, defines which policy to use:
+    - `UPDATE_POLICY_LATEST`: upgrade to latest version. 
+      - mutable: upgrade all packages to latest versions.
+      - immutable: upgrade the OS version to the latest.
+    - `UPDATE_POLICY_TARGET`: upgrade to a specific version.
+      - mutable: upgrade all packages to the specified version defined in 
+        the `install_package` list.
+      - immutable: upgrade the OS version to OS Version linked by the 
+        `target_os` edge.
+  - `target_os`: edge to the OS Profile that must be installed during the 
+    update. This field is used only for Immutable OSes.
+
+- **OS Update Job**: stores the information about the update job that run 
+  on the Edge Node. This resource is created when the update is started, 
+  and stores information about what happened in the Edge Node during the 
+  Update.
+  - `status`: stores the short description of what happened during the update.
+  - `status_indicator`: enum field, moder status indicator:
+    - `STATUS_INDICATION_ERROR`: an error happened during the update.
+    - `STATUS_INDICATION_IN_PROGRESS`: unused, the progress about the 
+      update is kept into the `update_status` field of the Instance.
+    - `STATUS_INDICATION_IDLE`: the update was successful.
+  - `status_timestamp`: timestamp of the last status update. Overlaps with 
+    the `end_time`, but it's kept for consistency with Modern Status. Also, 
+    could be used in the future, if we deprecate the Instance `update_status`.
+  - `status_details`: detailed description of what happend during the 
+    update. For example, the list of packages that were installed or 
+    updated, or the reason why the Immutable OS update failed.
+  - `start_time`: timestamp of when the update job started.
+  - `end_time`: timestamp of when the update job ended.
+  - `applied_policy`: the policy that was applied during the update. 
+    Provides information of what was the target Update Policy for this update.
+
+Resource and Field Handling:
+
+- **OS Profile**:
+  - `installed_packages`: this field is handled and filled by OSRM also for 
+    Mutable OSes.
+- **Instance**:
+  - `current_os`: after provisioning, this field is handled internally by 
+    EIM. The Maintenance Manager (MM) keeps it updated after the 
+    maintenance schedule happened.
+  - `installed_packages`: this field is handled by Host Resource Manager (HRM)
+    that received the information about installed package by the Hardware 
+    Discovery Manager (HDA).
```

Comment:
Why split tasks around update/packages across HRM and MM?  This would mean multiple agents on the EN talking to similar components (Turtle Creek, others), which could get into race conditions - what happens when HDA asks for packages installed when Update Agent is performing an update?

Reply:
Good point, for sure if we need to interact with the same tool used for update, I would keep that in PUA and MM.

Reply:
I agree we can make it part of PUA